committee/committee.go

```diff
@@ -165,39 +165,44 @@ func NewCommittee(kvstore db.KVStore, cfg Config) (Committee, error) {
 		nextHeight:            cfg.GravityChainStartHeight,
 	}, nil
 }
-
-func (ec *committee) Start(ctx context.Context) (err error) {
+func (ec *committee) load(ctx context.Context) (err error) {
 	ec.mutex.Lock()
 	defer ec.mutex.Unlock()
 	if err := ec.db.Start(ctx); err != nil {
 		return errors.Wrap(err, "error when starting db")
 	}
-	if startHeight, err := ec.db.Get(db.NextHeightKey); err == nil {
-		zap.L().Info("restoring from db")
-		ec.nextHeight = util.BytesToUint64(startHeight)
-		for height := ec.startHeight; height < ec.nextHeight; height += ec.interval {
-			zap.L().Info("loading", zap.Uint64("height", height))
-			data, err := ec.db.Get(ec.dbKey(height))
-			if err != nil {
-				return err
-			}
-			r := &types.ElectionResult{}
-			if err := r.Deserialize(data); err != nil {
-				return err
-			}
-			ec.cache.insert(height, r)
-			if err := ec.heightManager.add(height, r.MintTime()); err != nil {
-				return err
-			}
+	startHeight, err := ec.db.Get(db.NextHeightKey)
+	if err != nil {
+		return
+	}
+	zap.L().Info("restoring from db")
+	ec.nextHeight = util.BytesToUint64(startHeight)
+	for height := ec.startHeight; height < ec.nextHeight; height += ec.interval {
+		zap.L().Info("loading", zap.Uint64("height", height))
+		data, err := ec.db.Get(ec.dbKey(height))
+		if err != nil {
+			return err
+		}
+		r := &types.ElectionResult{}
+		if err := r.Deserialize(data); err != nil {
+			return err
+		}
+		ec.cache.insert(height, r)
+		if err := ec.heightManager.add(height, r.MintTime()); err != nil {
+			return err
 		}
 	}
+	return nil
+}
+func (ec *committee) Start(ctx context.Context) (err error) {
+	ec.load(ctx)
 	zap.L().Info("catching up via network")
 	tipHeight, err := ec.carrier.TipHeight()
 	if err != nil {
 		return errors.Wrap(err, "failed to get tip height")
 	}
-	results, errs := ec.fetchInBatch(tipHeight)
-	if err := ec.storeInBatch(results, errs); err != nil {
+	err = ec.fetchInBatchSplit(tipHeight)
```

Comment:
Can we not to do the fetch at all?

The sync will happen anyway after start and when having subscribed an eth block, right?